hack/setup.sh

```diff
@@ -0,0 +1,129 @@
+
+echo "Waiting for all Knative Eventing pods to be in the Running or Completed state..."
+
+while IFS= read -r line; do
+    POD_NAME=$(echo $line | awk '{print $1}')
+    POD_STATUS=$(echo $line | awk '{print $3}')
+    if [ "$POD_STATUS" != "Running" ] && [ "$POD_STATUS" != "Completed" ]; then
+        echo "Waiting for pod $POD_NAME to reach Running or Completed state..."
+        sleep 5
+        kubectl get pods -n knative-eventing | grep -v NAME | while IFS= read -r line; do continue; done
+    fi
+done < <(kubectl get pods -n knative-eventing | grep -v NAME)
+
+echo "All Knative Eventing pods are in the desired state."
+
+echo "<Success!>Knative Eventing Installation Completed Successfully!"
+
+
+echo "Before we proceed with installing the config, we need to check whether ko is installed or not."
+
+if command -v ko >/dev/null 2>&1; then
+    echo "ko is installed, proceeding with the installation of the config."
+else
+    echo "ko is not installed. Installing ko..."
+    # Check if Go is installed
+    if ! command -v go >/dev/null 2>&1; then
+        echo "Go is not installed. Please manually install Go first."
+        exit 1
+    fi
+
+    # Install ko
+    echo "Installing ko..."
+    GO111MODULE=on go get github.com/google/ko/cmd/ko
+
+    # Verifying ko installation
+    if command -v ko >/dev/null 2>&1; then
+        echo "ko successfully installed. Proceeding with the installation of the config"
+    else
+        echo "Failed to install ko. Please try to manually install ko."
+        exit 1
+    fi
+fi
+
+echo "Installing the config..."
+ko apply -f config/
+
+echo "Waiting for all pods to be in the Running or Completed state..."
+
+while IFS= read -r line; do
+    POD_NAME=$(echo $line | awk '{print $1}')
+    POD_STATUS=$(echo $line | awk '{print $3}')
+    if [ "$POD_STATUS" != "Running" ] && [ "$POD_STATUS" != "Completed" ]; then
+        echo "Waiting for pod $POD_NAME to reach Running or Completed state..."
+        sleep 5
+        kubectl get pods -n knative-eventing | grep -v NAME | while IFS= read -r line; do continue; done
+    fi
+done < <(kubectl get pods -n knative-eventing | grep -v NAME)
+
+echo "All the pods are in the desired state."
+
+echo "<Success!> Config Installation Completed Successfully!"
+
+echo "Install the sample source and event display"
+kubectl apply -f example/
+
+echo "Waiting for all pods to be in the Running or Completed state..."
+
+while IFS= read -r line; do
+    POD_NAME=$(echo $line | awk '{print $1}')
+    POD_STATUS=$(echo $line | awk '{print $3}')
+    if [ "$POD_STATUS" != "Running" ] && [ "$POD_STATUS" != "Completed" ]; then
+        echo "Waiting for pod $POD_NAME to reach Running or Completed state..."
+        sleep 5
+        kubectl get pods -n knative-eventing | grep -v NAME | while IFS= read -r line; do continue; done
+    fi
+done < <(kubectl get pods -n knative-eventing | grep -v NAME)
```

Comment:
```suggestion
done < <(kubectl get pods -n knative-samples | grep -v NAME)
```
The example is in a different namespace :)